types/types.ts

```diff
@@ -4,22 +4,25 @@ export interface WerkitMeta {
   runtime_info?: any
 }
 
-export interface WerkitSuccessResult<ResultType> extends WerkitMeta {
+export interface WerkitSuccessResult<ResultType, MessageKeyType>
```

Comment:
The only thing I'm concerned about with `MessageKeyType` is the ability for functions to generate JSON schema for their responses. I guess the types they export will have to use `MessageKeyType: any`… and then the caller could generate a new, more specific type + schema if they want to. Does that sound right?

Reply:
>  I guess the types they export will have to use MessageKeyType: any… and then the caller could generate a new, more specific type + schema if they want to. Does that sound right?

Yes. You can see how this is done in vogue types package, and goldie-data-layer. First, [here](https://github.com/curvewise/vogue/pull/46/files#diff-95ad1cbdb5cbb2abcced4aac70f142ed53e87f009151b3a457c3d1186e50fa19R24) is the more specific Werkit result type for vogue, called VogueResult, which is made more specific by specifying the MessageKeyType. This is used to generate JSON schema [here](https://github.com/curvewise/vogue/pull/46/files#diff-f61f9c533761b8c275639e5ff910f4ebd6b9e43867e8ff79ec99742c453dfa55R133), which is then used by goldie-data-layer [here](https://github.com/curvewise/goldie-data-layer/pull/143/files#diff-49f926daa14e9b17f86bf2ece5ad30f1e3ed7175945f879d117e7b0e93c346c5R17) to validate its graphile-worker payload.

As you mentioned [here](https://github.com/curvewise/vogue/pull/46#discussion_r571143379), I will adjust this by moving the MessageKey definition from vogue types package into goldie-data-layer